initialization/initialization.go

```diff
@@ -44,61 +42,90 @@ var (
 	ErrStateMetadataFileMissing     = errors.New("metadata file is missing")
 )
 
+// Providers returns a list of available compute providers.
 func Providers() []ComputeProvider {
 	return gpu.Providers()
 }
 
-func CPUProviderID() int {
+// CPUProviderID returns the ID of the CPU provider or nil if the CPU provider is not available.
+func CPUProviderID() uint {
 	return gpu.CPUProviderID()
 }
 
-type initializeOption struct {
-	commitment []byte
-	cfg        *Config
-	initOpts   *config.InitOpts
-	logger     Logger
+type option struct {
+	nodeId []byte
+	atxId  []byte
+
+	cfg      *Config
+	initOpts *config.InitOpts
+	logger   Logger
 }
 
-func (opts *initializeOption) verify() error {
-	if opts.cfg == nil {
+func (o *option) verify() error {
+	if o.nodeId == nil {
+		return errors.New("`nodeId` is required")
+	}
+
+	if o.atxId == nil {
+		return errors.New("`atxId` is required")
+	}
+
+	if o.cfg == nil {
 		return errors.New("no config provided")
 	}
 
-	if opts.initOpts == nil {
+	if o.initOpts == nil {
 		return errors.New("no init options provided")
 	}
 
-	return config.Validate(*opts.cfg, *opts.initOpts)
+	return config.Validate(*o.cfg, *o.initOpts)
 }
 
-type initializeOptionFunc func(*initializeOption) error
+type OptionFunc func(*option) error
 
-func WithCommitment(commitment []byte) initializeOptionFunc {
-	return func(opts *initializeOption) error {
-		if len(commitment) != 32 {
-			return fmt.Errorf("invalid `id` length; expected: 32, given: %v", len(commitment))
+// WithNodeId sets the ID of the Node.
+func WithNodeId(nodeId []byte) OptionFunc {
+	return func(opts *option) error {
+		if len(nodeId) != 32 {
+			return fmt.Errorf("invalid `id` length; expected: 32, given: %v", len(nodeId))
 		}
-		opts.commitment = commitment
+
+		opts.nodeId = nodeId
+		return nil
+	}
+}
+
+// WithAtxId sets the ID of the CommitmentATX.
```

Comment:
Not critical, but unlike `Initializer`, the work oracle doesn't need to know about `atxId` and `nodeId`, but rather only about `commitment`. This design also adds a few unnecessary hashes (per batch invocation, instead of a single one in `Initializer`).

Reply:
Moved `CommitmentBytes` function to `shared` package. `WorkOracle` now takes a `WithCommitment` functional option, while callers of the oracle use the `CommitmentBytes` function to calculate the commitment from the Node ID and Commitment ATX ID